airflow/utils/log/logging_mixin.py

```diff
@@ -38,17 +39,15 @@ class LoggingMixin:
     """Convenience super-class to have a logger configured with the class name"""
 
     def __init__(self, context=None):
+        self._log: Optional[logging.Logger] = None
         self._set_context(context)
 
     @property
     def log(self) -> Logger:
         """Returns a logger."""
-        try:
-            # FIXME: LoggingMixin should have a default _log field.
-            return self._log  # type: ignore
-        except AttributeError:
+        if not hasattr(self, '_log') or self._log is None:
```

Comment:
Why is the `hasattr()` check needed?

Reply:
Some of the classes that inherit from LoggingMixin don't call `super().__init__()`. 
I counted above 5 of them.
For that cases `hasattr` is needed. 
I assume that `or self._log is None` could be deleted.

Reply:
Yep. The logging Mixin is not consistently used. I think it's safer to leave it here I think as it's rather likely there will be other cases where init() is not caled for the Mixin.

Reply:
How about making `_log` a class variable instead?

```python
class LoggingMixin:
    _log: Optional[logging.Logger] = None

    def __init__(self, context=None):
        self._set_context(context)

    @property
    def log(self) -> Logger:
        """Returns a logger."""
        if self._log is None:
            self._log = ...
        return self._log
```

I think this should work.

Reply:
Done

The advantage of this solution will be the absence of usage `hasattr` function and all child classes will have `_log` attribute.

Firstly I thought that if the child class doesn't use `.log` then `_log` attribute just will allocate memory useless. But then I realized that I don't see a reason to inherit from `LoggingMixin` and not use `.log`.

Reply:
Class attributes are not allocated for each subclass, only once for the parent class that defines it, so subclasses would have no additional allocations to choose waste, even if they do inherit LoggingMixin withou using `log`.